includes/config.inc

```diff
@@ -540,6 +548,74 @@ function islandora_solr_metadata_config_field_form($form, &$form_state, $config_
 }
 
 /**
+ * Generate the element definition for the permission fieldset.
+ *
+ * @param array $permissions
+ *   The selected permissions array containing
+ *   key values:
+ *     'enable_permissions' => bool (enable/disable permissions),
+ *     'permissions' => array (an array of permissions to apply)
+ * @param array $permissions_default
+ *   The permissions to select by default.
+ * @param array $permissions_disable
+ *   Some permissions which we will not allow to be changed (set as disabled in
+ *   the form).
+ *
+ * @return array
+ *   An associative array containing the definition for the permissions
+ *   fieldset.
+ */
+function islandora_solr_metadata_get_admin_permissions_fieldset($permissions, $permissions_default, $permissions_disable) {
+  $permissions_fieldset = array(
+    '#type' => 'fieldset',
+    '#title' => t('Permissions'),
+    '#collapsible' => TRUE,
+    '#collapsed' => TRUE,
+    'enable_permissions' => array(
+      '#type' => 'checkbox',
+      '#title' => t('Enable permissions'),
+      '#default_value' => $permissions['enable_permissions'],
+    ),
+    'permissions' => array(
+      '#type' => 'checkboxes',
+      '#title' => t('Permissions'),
+      '#options' => user_roles(),
+      '#default_value' => !empty($permissions['permissions']) ? $permissions['permissions'] : $permissions_default,
+      '#description' => t('Select which roles can access this field.<br /><strong>Note:</strong> checkboxes may be grayed out for roles which do not have permission to search the Solr index.'),
+      '#states' => array(
+        'visible' => array(
+          ':input[name="wrapper[permissions][enable_permissions]"]' => array('checked' => TRUE),
+        ),
+      ),
+    ),
+  );
+  foreach ($permissions_disable as $rid) {
+    $permissions_fieldset['permissions'][$rid] = array(
+      '#disabled' => TRUE,
+    );
+  }
+
+  return $permissions_fieldset;
+}
+
+/**
+ * Utility function to append permissions and actions to the modal.
+ *
+ * @param array $values
```

Comment:
[Out of sync](https://travis-ci.org/Islandora/islandora_solr_metadata/jobs/176811814#L365):
```
+604: [minor] Doc comment for var $values does not match actual variable name $permissions at position 1
```